src/vm-vcpu-ref/src/x86_64/cpuid.rs

```diff
@@ -15,11 +15,32 @@ const ECX_TSC_DEADLINE_TIMER_SHIFT: u32 = 24; // TSC deadline mode of APIC timer
 const ECX_HYPERVISOR_SHIFT: u32 = 31; // Flag to be set when the cpu is running on a hypervisor.
 const EDX_HTT_SHIFT: u32 = 28; // Hyper Threading Enabled.
 
-/// Filter CPUID.
+/// Updates the passed `cpuid` such that it can be used for configuring a vCPU
+/// for running.
+///
+/// # Example
+///
+/// We are recommending the `cpuid` to be created from the supported CPUID on
+/// the running host.
+///
+/// ```rust
+/// use kvm_bindings::CpuId;
+/// use kvm_ioctls::{Error, Kvm};
+/// use vm_vcpu_ref::x86_64::cpuid::filter_cpuid;
+///
+/// fn default_cpuid(cpu_index: usize, num_vcpus: usize) -> Result<CpuId, Error> {
+///     let kvm = Kvm::new()?;
+///     let mut cpuid = kvm.get_supported_cpuid(kvm_bindings::KVM_MAX_CPUID_ENTRIES)?;
+///     filter_cpuid(&kvm, cpu_index, num_vcpus, &mut cpuid);
+///     Ok(cpuid)
+/// }
+///
+/// # default_cpuid(0, 1).unwrap();
+/// ```
 pub fn filter_cpuid(kvm: &Kvm, vcpu_id: usize, cpu_count: usize, cpuid: &mut CpuId) {
```

Comment:
Would it be better to have vcpu_id and cpu_count as u8 for example, so we would avoid possible unsafe `as u32`s below or isn't truncating the values a problem here?

Reply:
Done.